src/server/array_func.c

```diff
@@ -380,6 +380,11 @@ update_array_indices_remaining_attr(job *parent)
 {
 	struct ajtrkhd	*ptbl = parent->ji_ajtrk;
 
+	if (ptbl->tkm_flags & TKMFLG_RUN_JOB_REQ) {
+		ptbl->tkm_flags &= ~TKMFLG_RUN_JOB_REQ;
```

Comment:
it seems like req_runjob has the capability to run a range of subjobs. By removing the flag here, it seems like this will only work for the first subjob in a range. So, I think we should not remove the flag here, what do you think?

Reply:
I ran a 40k job array test, I didn't see a single occurrence of a range of sub-jobs to run job request. Looks like this is not a supported feature. In the future, if we add support for it, the functionality should be added with full optimization. It's not required today.

Reply:
@bremanandjk 'req_runjob' is also used for qrun, which supports range of subjob, run same test with sched off and do qrun of range of subjob

Reply:
Yeah, for supporting those user-triggered corner cases we would need to add code logic, that may burden the scheduler triggered run path. What do you think? Is there a way to separate out both the paths. I feel It's not a common path to consider. It's my opinion.

Reply:
There are two cases here.  The admin can do a qrun 123[1-10] saying they want to run those subjobs.  That is then passed to the scheduler.  The scheduler has code to handle this request.  The second is a qrun -H.  This is how the admin overrides the scheduler and does what they want to do.  We shouldn't break either of these.

Reply:
Thanks for your comment Bhroam, just wanted to point that the proposed change won't break the range of sub-jobs functionality, this optimization would not be applied to a range of sub-jobs path. 

Anyhow, further discussion converged not to remove the attribute computation in the run path. There won't be any difference between the single sub job and range of sub-jobs then.